lib/mayaHydra/hydraExtensions/mhLeadObjectPathTracker.cpp

```diff
@@ -0,0 +1,169 @@
+    {
+        public:
+            GlobalSelectionChangedObs(MayaHydra::MhLeadObjectPathTracker& leadObjectPathInstance)
+            : _leadObjectPathInstance(leadObjectPathInstance)
+            {
+            }
+
+        private:
+            void operator()(const Ufe::Notification& notification) override
+            {
+                const Ufe::SelectionChanged& selectionChanged = notification.staticCast<Ufe::SelectionChanged>();
+
+                auto leadObjectUfePath = _leadObjectPathInstance.getLeadObjectUfePath();
+
+                switch (selectionChanged.opType()) {
+                    case Ufe::SelectionChanged::OpType::Append: {
+                        const Ufe::SelectionItemAppended& appended = notification.staticCast<Ufe::SelectionItemAppended>();
+                        auto newLeadObjectUfePath = appended.item()->path();
+                        _leadObjectPathInstance.setNewLeadObjectSceneItem(newLeadObjectUfePath);
+                        break;
+                    }
+                    case Ufe::SelectionChanged::OpType::Remove: {
+                        const Ufe::SelectionItemRemoved& removed = notification.staticCast<Ufe::SelectionItemRemoved>();
+                        auto removedItemUfePath = removed.item()->path();
+                        //Check if the lead object has been removed
+                        if (leadObjectUfePath == removedItemUfePath) {
+                            UpdateLeadObjectSceneItem(leadObjectUfePath);
+                        }
+                        break;
+                    }
+                    case Ufe::SelectionChanged::OpType::Clear: {
+                        _leadObjectPathInstance.ClearLeadObject();
+                        break;
+                    }
+                    
+                    case Ufe::SelectionChanged::OpType::Insert://Fall into
+                    case Ufe::SelectionChanged::OpType::SelectionCompositeNotification://Fall into
+                    case Ufe::SelectionChanged::OpType::ReplaceWith: {
+                        UpdateLeadObjectSceneItem(leadObjectUfePath);
+                        break;
+                    }
+                    default: break;
+                }
+            }
+
+            void UpdateLeadObjectSceneItem(const Ufe::Path& currentLeadObjectUfePath) { 
+                //Update the lead object if it has changed
+                auto globalSelection = Ufe::GlobalSelection::get();
+                if (globalSelection->size() > 0) {
+                    auto newLeadObjectUfePath = globalSelection->back()->path();
+                    if (newLeadObjectUfePath != currentLeadObjectUfePath){
+                        _leadObjectPathInstance.setNewLeadObjectSceneItem(newLeadObjectUfePath);
+                    }
+                }else {
+                    _leadObjectPathInstance.ClearLeadObject();
+                }
+            }
+
+            MayaHydra::MhLeadObjectPathTracker& _leadObjectPathInstance;
```

Comment:
Confusing name, how about
```suggestion
            MayaHydra::MhLeadObjectPathTracker& _leadObjectPathTracker;
```